explicit_bzero Warning on Linux

Building libcrux on linux I get the following warnings: 

```bash
$ cargo build
warning: libcrux-hacl@0.0.2-beta.3: /libcrux/sys/hacl/c/src/Lib_Memzero0.c: In function 'Lib_Memzero0_memzero0':
warning: libcrux-hacl@0.0.2-beta.3: /libcrux/sys/hacl/c/src/Lib_Memzero0.c:48:6: warning: #warning "Your platform does not support any safe implementation of memzero -- consider a pull request!" [-Wcpp]
warning: libcrux-hacl@0.0.2-beta.3:    48 |     #warning "Your platform does not support any safe implementation of memzero -- consider a pull request!"
warning: libcrux-hacl@0.0.2-beta.3:       |      ^~~~~~~
warning: libcrux-hacl@0.0.2-beta.3: /libcrux/sys/hacl/c/src/Lib_Memzero0.c:36:10: warning: unused variable 'len_' [-Wunused-variable]
warning: libcrux-hacl@0.0.2-beta.3:    36 |   size_t len_ = (size_t) len;
warning: libcrux-hacl@0.0.2-beta.3:       |          ^~~~
   Compiling libcrux v0.0.2-beta.3 (/libcrux)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.44s
```

This is caused because `LINUX_NO_EXPLICIT_BZERO` is set [here](https://github.com/cryspen/libcrux/blob/9a7b4f028ed657b76dba20ba10bc96419b9b7d3a/sys/hacl/build.rs#L257). 

However `explicit_bzero` is available on my system (I compiled a simple program using it for verification) and I am using the nix flake anyways. Removing the line line in `sys/hacl/c/src/Lib_Memzero0.c` removes the warning and all tests still pass. 

Is there a reason why `LINUX_NO_EXPLICIT_BZERO` is always set? 

Comment says
> We disable this to avoid running into issues

Maybe use 
```c
#define _GNU_SOURCE
defined(__GLIBC__) && (__GLIBC__ >= 2) && (__GLIBC_MINOR__ >= 25)
```

--- a/sys/hacl/build.rs
+++ b/sys/hacl/build.rs
@@ -254,7 +254,6 @@ fn build(platform: &Platform, home_path: &Path) {
     }
     let hacl_target_arch = platform.hacl_target_arch.to_string();
     defines.push(("TARGET_ARCHITECTURE", hacl_target_arch.as_str()));
-    defines.push(("LINUX_NO_EXPLICIT_BZERO", "1")); // We disable this to avoid running into issues.
 
     // Platform detection
     if platform.simd128 {

--- a/sys/hacl/c/src/Lib_Memzero0.c
+++ b/sys/hacl/c/src/Lib_Memzero0.c
@@ -39,7 +39,9 @@ void Lib_Memzero0_memzero0(void *dst, uint64_t len) {
     SecureZeroMemory(dst, len);
   #elif defined(__APPLE__) && defined(__MACH__)
     memset_s(dst, len_, 0, len_);
-  #elif (defined(__linux__) && !defined(LINUX_NO_EXPLICIT_BZERO)) || defined(__FreeBSD__)
+  #elif defined(__linux__) && defined(__GLIBC__) && (__GLIBC__ >= 2) && (__GLIBC_MINOR__ >= 25)
+    explicit_bzero(dst, len_);
+  #elif defined(__FreeBSD__)
     explicit_bzero(dst, len_);
   #elif defined(__NetBSD__)
     explicit_memset(dst, 0, len_);

